linkerd/app/inbound/src/metrics/authz.rs

```diff
@@ -0,0 +1,153 @@
+use crate::policy::{AllowPolicy, Permit};
+use linkerd_app_core::{
+    metrics::{metrics, AuthzLabels, Counter, FmtMetrics, ServerLabel},
+    transport::labels::TargetAddr,
+};
+use parking_lot::RwLock;
+use std::{collections::HashMap, sync::Arc};
+
+metrics! {
+    inbound_http_authz_allow_total: Counter {
+        "The total number of inbound HTTP requests that were authorized"
+    },
+    inbound_http_authz_deny_total: Counter {
+        "The total number of inbound HTTP requests that could not be processed due to a proxy error."
+    },
+
+    inbound_tcp_authz_allow_total: Counter {
+        "The total number of inbound HTTP requests that were authorized"
+    },
+    inbound_tcp_authz_deny_total: Counter {
+        "The total number of inbound HTTP requests that were denied"
+    },
+    inbound_tcp_authz_terminate_total: Counter {
+        "The total number of inbound HTTP requests that were terminated due to an authorization change"
+    }
+}
+
+#[derive(Clone, Debug, Default)]
+pub(crate) struct HttpAuthzMetrics(Arc<HttpInner>);
+
+#[derive(Clone, Debug, Default)]
+pub(crate) struct TcpAuthzMetrics(Arc<TcpInner>);
+
+#[derive(Debug, Default)]
+struct HttpInner {
+    allow: RwLock<HashMap<(TargetAddr, AuthzLabels), Counter>>,
+    deny: RwLock<HashMap<(TargetAddr, ServerLabel), Counter>>,
+}
+
+#[derive(Debug, Default)]
+struct TcpInner {
+    allow: RwLock<HashMap<(TargetAddr, AuthzLabels), Counter>>,
+    deny: RwLock<HashMap<(TargetAddr, ServerLabel), Counter>>,
+    terminate: RwLock<HashMap<(TargetAddr, ServerLabel), Counter>>,
```

Comment:
AFAICT, these are never actually read concurrently --- it looks like every metrics update requires a write lock. Therefore, we're not really gaining anything from using a `RwLock` rather than a `Mutex` here...in fact, it may be a bit less efficient.

Unless we're planning to change the code that updates the metrics to try to increment with a read lock, and only acquire the write lock when we need to insert a new set of labels, we should *probably* just replace these with `Mutex`es).